crates/tako/src/worker/rpc.rs

```diff
@@ -358,6 +354,12 @@ async fn worker_message_loop(
                     .insert(msg.worker_id, msg.address)
                     .is_none())
             }
+            ToWorkerMessage::Reservation(on_off) => {
+                state.reservation = on_off;
+                if !on_off {
+                    state.last_task_finish_time = Instant::now();
```

Comment:
Now that it's reset on more places, I would create a method like `reset_idle_timer()`.